lib/dash/dash_parser.js

```diff
@@ -1507,6 +1548,92 @@ shaka.dash.DashParser = class {
     }
   }
 
+  /**
+   * @param {!Array<shaka.extern.Period>} periods
+   * @private
+   */
+  prunePeriodCache_(periods) {
+    const activeIds = new Set(periods.map((period) => period.id));
+    for (const id of Array.from(this.periodCache_.keys())) {
+      if (!activeIds.has(id)) {
+        this.periodCache_.delete(id);
+      }
+    }
+  }
+
+  /**
+   * @param {?string} periodId
+   * @param {?number} start
+   * @return {string}
+   * @private
+   */
+  getPeriodCacheKey_(periodId, start) {
+    if (periodId) {
+      return periodId;
+    }
+    return '__shaka_period_' + (start != null ? start : 'null');
+  }
+
+  /**
+   * Simple FNV-1a hash implementation
+   * @param {string} str
+   * @param {number} seed
+   * @return {number}
+   * @private
+   */
+  fnv1a_(str, seed) {
```

Comment:
I would like to see these (fnv1a_ and hashTXml_) moved to a utility class.  It's distracting here in the DASH parser, and also not testable as a private method.